tests/test_nodes.py

```diff
@@ -279,6 +280,15 @@ def test_as_string_unknown() -> None:
         assert nodes.Unknown().as_string() == "Unknown.Unknown()"
         assert nodes.Unknown(lineno=1, col_offset=0).as_string() == "Unknown.Unknown()"
 
+    @staticmethod
+    def test_recursion_error_trapped() -> None:
+        with pytest.warns(UserWarning, match="unable to transform"):
+            ast = abuilder.string_build(LONG_CHAINED_METHOD_CALL)
+
+        attribute = ast.body[1].value.func
+        with pytest.raises(UserWarning):
+            attribute.as_string()
```

Comment:
absolutely, but it seems pytest is doing something magical if the using same recursion limit is not a problem in the interactive interpreter but is a problem inside pytest?

Reply:
Yeah, pytest and warnings capture is just magic sometimes 😅
Since the test pass an it works in pylint I would be fine with merging this as is (even with the one uncovered line).